spec/lazy/directory_spec.rb

```diff
@@ -0,0 +1,169 @@
+require_relative "../spec_helper"
+require "fauxpaas/lazy/directory"
+require "pathname"
+require "fileutils"
+require "find"
+
+module Fauxpaas
+  RSpec.describe Lazy::Directory do
+    class TestFile
+      attr_reader :path
+      def initialize(path)
+        @path = Pathname.new(path)
+      end
+
+      def cp(dest)
+        self.class.new(dest)
+      end
+
+      def relative_from(base)
+        self.class.new(path.relative_path_from(base))
+      end
+
+      def merge(other)
+        self.class.new(path/"MERGED")
+      end
+
+      def eql?(other)
+        path == other.path
+      end
+      alias_method :==, :eql?
+    end
+
+    let(:files) do
+      [
+        TestFile.new(path/"bar.txt"),
+        TestFile.new(path/"collide.txt"),
+      ]
+    end
+    let(:base_path) { Pathname.new("lhs") }
+    let(:path) { base_path/"foo" }
+    let(:directory) { described_class.new(path, files) }
+
+    describe "::for" do
+      before(:each) { allow(Find).to receive(:find).and_return(raw_files) }
+      let(:raw_files) do
+        [
+          "#{path}/bar.txt",
+          "#{path}/collide.txt"
+        ]
+      end
+
+      it "wraps the path in a pathname" do
+        expect(described_class.for(path).path).to eql(Pathname.new(path))
+      end
+
+      it "creates an instance with path" do
+        expect(described_class.for(path).path).to eql(path)
+      end
+
+      xit "stores the files" do
```

Comment:
Why is this one skipped?

Reply:
To actually function it requires either mocking the filesystem or setting up a sandbox.